### PR TITLE
chore: linting for no exported subjects

### PR DESCRIPTION
**Pull request by davidyoung77**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Styling and layout changes
- [x] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

- rxjs linting rules improvments

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/shared/ui/nuverial/src/lib/components/snackbar/snackbar.component.ts [hunk 1]
```diff
@@ -100,19 +100,34 @@ export class NuverialSnackbarComponent {
...
+
+  private readonly _action: Subject<unknown> = new Subject<unknown>();
+  private readonly _closed: Subject<void> = new Subject<void>();
 
   constructor(
     @Inject(MAT_SNACK_BAR_DATA)
     public nuverialSnackBarConfiguration: INuverialSnackBarComponentOptions,
-  ) {}
+  ) {
+    this.action = this._action.asObservable();
```
Comment (thechechogomez) [inline libs/shared/ui/nuverial/src/lib/components/snackbar/snackbar.component.ts:118]: Is there a difference in assigning the observables on the constructor or initializing it as the observable of the subject? If we want to go for consistency, I think in most places initialize the property as the observable of the subject.

**Review by thechechogomez — APPROVED** [review]
Looks good, just have a question.

File: libs/shared/ui/nuverial/src/lib/components/snackbar/snackbar.component.ts [hunk 2]
```diff
@@ -100,19 +100,34 @@ export class NuverialSnackbarComponent {
    * Action event emits action context when action element clicked
    * @ignore
    */
-  public action: Subject<unknown> = new Subject<unknown>();
+  public action: Observable<unknown>;
```
Comment (kevincheng0) [inline libs/shared/ui/nuverial/src/lib/components/snackbar/snackbar.component.ts:103]: should we have `action$` instead to match the observable naming convention?

File: libs/shared/ui/nuverial/src/lib/components/snackbar/snackbar.component.ts (hunk 1 above)
Comment (davidyoung77) [reply libs/shared/ui/nuverial/src/lib/components/snackbar/snackbar.component.ts:118]: I was debating that functionally it is the same but it is good practice to to sort public properties above private ones which enforces this implementation, our linting would enforce this but it gets overridden unfortunately,

I'll think about this for a minute more

File: libs/shared/ui/nuverial/src/lib/components/snackbar/snackbar.component.ts (hunk 2 above)
Comment (davidyoung77) [reply libs/shared/ui/nuverial/src/lib/components/snackbar/snackbar.component.ts:103]: Yep will update

**Review by kevincheng0 — APPROVED** [review]

**Comment by davidyoung77** [comment]
Merging despite SQ complaint this did not add any code just changed names

**Pull request merged**